manifests/a/Apache/Maven/3.9.5/Apache.Maven.installer.yaml

```diff
@@ -0,0 +1,19 @@
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.5.0.schema.json
+
+PackageIdentifier: Apache.Maven
+PackageVersion: 3.9.5
+MinimumOSVersion: 10.0.0.0
+InstallerType: zip
```

Comment:
In case of Maven and Java I would really not go that way. This will install an JDK what the person maybe do not want and which could interfere with an already installed version. Often developers have already either an Oracle, Eclipse Adoptium and so on installed. I would leave it to the people to decide on their own which one.

Reply:
That's a fair point. Maybe instead of adding this dependency, we can mention this in `InstallationNotes` that a valid java installation is required which would show this message to the user post-installation.

cc @denelon @Trenly - What would be your thoughts on this one? On one hand the package may not work properly on a fresh machine if we don't specify the dependency but on the flip side adding it may mess up existing installations of Java for users that already have the dependency. I'm not sure which scenario is more common and also not sure if WinGet can pick up every time that the dependency is present, since this dependency can be fulfilled by a number of different packages i.e., Oracle.JDK, EclipseAdoptium.Temurin.JDKs ...

This looks related to https://github.com/microsoft/winget-cli/issues/2969

Reply:
I've added `InstallationNotes` accordingly with a hint to run `winget install Microsoft.OpenJDK.21` otherwise.

Reply:
Yeah, we've had lots of discussions about runtimes that may come from any number of providers. The "ExternalDependency" was initially conceived to display some information about an external dependency like Java. We're looking at a few different options to resolve these kinds of dependencies.

I like the installation notes method, but it's still going to cause a headache with automated validation.

Reply:
I see. What really might help, might be stuff, to execute something and check the result. If the expected criteria isn't met, install the specified dependency. But this is something for the future.

Reply:
We have been leaning towards a type of "provider" model. The idea is that for these kinds of dependencies where more than one different package may satisfy the dependency, we can have a "special" provider type of thing. Then the "manifest" for the provider can be updated to prevent the desire to update every single manifest with a "one of" type of mechanism. It also generates a list of packages a user could pick from to satisfy the dependency. In a "headless" scenario, we'll have to reason about the best logic to have a "default" provider to install if so specified, or a mechanism to "pass" the dependency package ID in to make a selection for automated scenarios to run headless.